tchannel/tchannel.py

```diff
@@ -319,3 +321,50 @@ def advertise(self, routers=None, name=None, timeout=None,
         response = Response(json.loads(body), headers or {})
 
         raise gen.Return(response)
+
+    @gen.coroutine
+    def stop(self, reason=None):
```

Comment:
Unadvertise is a call to hyperbahn. Stopping out unad loop means hyperbahn still thinks we're available for the next 5 minutes.